expression/builtin_json_vec.go

```diff
@@ -245,11 +245,79 @@ func (b *builtinJSONKeys2ArgsSig) vecEvalJSON(input *chunk.Chunk, result *chunk.
 }
 
 func (b *builtinJSONLengthSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONLengthSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	nr := input.NumRows()
+
+	jsonBuf, err := b.bufAllocator.get(types.ETJson, nr)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(jsonBuf)
+	if err := b.args[0].VecEvalJSON(b.ctx, input, jsonBuf); err != nil {
+		return err
+	}
+	result.ResizeInt64(nr, false)
+	resI64s := result.Int64s()
+
+	if len(b.args) == 2 {
+		pathBuf, err := b.bufAllocator.get(types.ETString, nr)
+		if err != nil {
+			return err
+		}
+		defer b.bufAllocator.put(pathBuf)
+		if err := b.args[1].VecEvalString(b.ctx, input, pathBuf); err != nil {
+			return err
+		}
+		result.MergeNulls(jsonBuf, pathBuf)
+		for i := 0; i < nr; i++ {
+			if result.IsNull(i) {
+				continue
+			}
+			jsonItem := jsonBuf.GetJSON(i)
+
+			if jsonItem.TypeCode != json.TypeCodeObject && jsonItem.TypeCode != json.TypeCodeArray {
```

Comment:
Why change the execution logic here? The order will affect the returned error.

Reply:
I have fixed the order problem. Now the logic is the same as the non-vec version.